src/serac/physics/thermal_solid_functional.hpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2019, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file thermal_solid_functional.hpp
+ *
+ * @brief An object containing an operator-split thermal structural solver
+ */
+
+#pragma once
+
+#include "mfem.hpp"
+
+#include "serac/physics/base_physics.hpp"
+#include "serac/physics/solid_functional.hpp"
+#include "serac/physics/thermal_conduction_functional.hpp"
+
+
+
+namespace serac {
+
+/**
+ * @brief The operator-split thermal-structural solver
+ *
+ * Uses Functional to compute action of operators
+ */
+template <int order, int dim>
+class ThermalSolidFunctional : public BasePhysics {
+public:
+  
+  /**
+   * @brief Construct a new coupled Thermal-Solid Functional object
+   *
+   * @param options The options for the linear, nonlinear, and ODE solves
+   * @param geom_nonlin Flag to include geometric nonlinearities
+   * @param keep_deformation Flag to keep the deformation in the underlying mesh post-destruction
+   * @param name An optional name for the physics module instance
+   */
+  ThermalSolidFunctional(const typename ThermalConductionFunctional<order, dim>::SolverOptions& thermal_options,
+                         const typename SolidFunctional<order, dim>::SolverOptions& solid_options,
+                         GeometricNonlinearities geom_nonlin = GeometricNonlinearities::On,
+                         FinalMeshOption keep_deformation = FinalMeshOption::Deformed,
+                         const std::string& name = "")
+      : BasePhysics(2, order),
```

Comment:
This should be `3` indicating 3 state fields (`displacement`, `velocity`, and `temperature`). However, I'm honestly not sure we use it anymore so I should probably just get rid of this argument.